scripts/troubleshoot/TroubleshootError.ps1

```diff
@@ -625,77 +657,142 @@ else {
-        Write-Host($message)
-        $question = " Do you want to onboard container health to the workspace?"
-
-        $choices = New-Object Collections.ObjectModel.Collection[Management.Automation.Host.ChoiceDescription]
-        $choices.Add((New-Object Management.Automation.Host.ChoiceDescription -ArgumentList '&Yes'))
-        $choices.Add((New-Object Management.Automation.Host.ChoiceDescription -ArgumentList '&No'))
-
-        $decision = $Host.UI.PromptForChoice($message, $question, $choices, 0)
-
-        if ($decision -eq 0) {
-            Write-Host("Deploying template to onboard container health : Please wait...")
-
-            $DeploymentName = "ContainerInsightsSolutionOnboarding-" + ((Get-Date).ToUniversalTime()).ToString('MMdd-HHmm')
-            $Parameters = @{ }
-            $Parameters.Add("workspaceResourceId", $LogAnalyticsWorkspaceResourceID)
-            $Parameters.Add("workspaceRegion", $WorkspaceLocation)
-            $Parameters
-            try {
-                New-AzResourceGroupDeployment -Name $DeploymentName `
-                    -ResourceGroupName $workspaceResourceGroupName `
-                    -TemplateUri  https://raw.githubusercontent.com/microsoft/Docker-Provider/ci_prod/scripts/onboarding/templates/azuremonitor-containerSolution.json `
-                    -TemplateParameterObject $Parameters -ErrorAction Stop`
+        try {
+            $dcrRuleName = "MSCI-" + $ClusterName + "-" + $ClusterRegion
+            $dcrRule = Get-AzDataCollectionRule -ResourceGroupName $workspaceResourceGroupName -RuleName $dcrRuleName -ErrorAction Stop -WarningAction silentlyContinue
+            Write-Host("Successfully fetched Data Collection Rule...") -ForegroundColor Green
+            $extensionNameInDCR =  $dcrRule.DataSources.Extensions.ExtensionName
+            if ($extensionNameInDCR -eq "ContainerInsights") {
+                $laDestinations = $dcrRule.Destinations.LogAnalytics
+                if (($null -ne $laDestinations) -and ($laDestinations.Length -gt 0) -and ($LogAnalyticsWorkspaceResourceID -eq $laDestinations[0].WorkspaceResourceId)) {
+                    Write-Host("Successfully validated Data Collection Rule is valid...") -ForegroundColor Green
+                } else {
+                    Write-Host("")
+                    Write-Host("Data Collection Rule: '" + $dcrRuleName + "' found is not valid ContainerInsights extension DCR.") -ForegroundColor Red
+                    Write-Host("")
+                    Stop-Transcript
+                    exit 1
+                }
 
+            } else {
+                Write-Host("")
+                Write-Host("Data Collection Rule: '" + $dcrRuleName + "' found is not valid ContainerInsights extension DCR.") -ForegroundColor Red
```

Comment:
should this error log be different from the one above? it will help in debugging

Reply:
> should this error log be different from the one above? it will help in debugging

>Good catch. Improved the log message.

Reply:
nit: do you think adding both the expected and actual log analytics workspace name in the log message will be helpful?

Reply:
> nit: do you think adding both the expected and actual log analytics workspace name in the log message will be helpful?

> fixed.